x/staking/keeper/keeper.go

```diff
@@ -33,6 +33,7 @@ type Keeper struct {
 	consensusAddressCodec addresscodec.Codec
 
 	Schema           collections.Schema
+	HistoricalInfo   collections.Map[[]byte, types.HistoricalInfo]
```

Comment:
use `uint64` as key. Reason is previously we converted the int64 into uint64 and then saved it in bigendian form, this is the cleanest solution. Anyways historical infos cannot be negative, so it's a mistake that the historical info type is int64 instead of being uint64. 

This also retains (state) backwards compatibility and allows us to remove keys function and all of that... it's going to be a little ugly because we're casting the int64 to uint64 everywhere, but I think it's better than having ` stakingtypes.HistoricalInfoKey` everywhere.

Cc @julienrbrt

Reply:
Yeah, thanks for the suggestion, just checked the backwards compatibility as well, no issues at all. Thanks!